src/main/java/uk/gov/hmcts/ccd/endpoint/exceptions/RestExceptionHandler.java

```diff
@@ -50,6 +60,29 @@ public RestExceptionHandler(AppInsights appInsights) {
                 .body(error);
     }
 
+    @ExceptionHandler(CaseValidationException.class)
+    @ResponseBody
+    public ResponseEntity<HttpError> handleCaseValidationException(HttpServletRequest request, CaseValidationException exception) {
+
+        // read field IDs from CaseValidationException.details (i.e. avoid using full error details as this may contain user data)
+        List<String> fieldIds = new ArrayList<>();
+        if (exception.getDetails() != null) {
+            String detailsJson = this.objectMapperService.convertObjectToString(exception.getDetails());
+            fieldIds = JsonPath.read(detailsJson, "$..field_errors[*].id");
+        }
+
+        Map<String, String> customProperties = new HashMap<>();
+        customProperties.put("CaseValidationError field IDs", Arrays.toString(fieldIds.toArray()));
+
+        LOG.warn("{}: The following list of fields are in an error state: {}", exception.getMessage(), fieldIds, exception);
```

Comment:
I was wondering whether it makes sense to not log the stacktrace for the the validation errors. What do you think @mattnayler?

Reply:
The same field level validation routines may be triggered in three separate places for a single UI call: 
* Initial event data submission
* After callback has returned new data
* After data has been merged with existing unchanged case details (i.e. before case event is saved)

So whilst the data submitted to AppInsights will always contain the full stack details it may still be useful for a dev to see their stack trace is different to the one they are investigating (i.e. it was for me during RDM-7840).  So I am currently tempted to leave it but also happy to be overruled, @mario-paniccia?

Reply:
right, good point. Let's leave it then cheers